From: Maximilian Weigand <mweigand@mweigand.net>
Date: Fri, 10 Mar 2023 10:07:00 +0100
Subject: [dts pinenote.dtsi] switch touchscreen (cyttsp5) interrupt from
 LEVEL LOW to EDGE_FALLING.

My notes:

- I think the interrupt should indeed be FALLING EDGE, and the level low
  interrupt type documented here:

https://github.com/torvalds/linux/blob/63355b9884b3d1677de6bd1517cd2b8a9bf53978/Documentation/devicetree/bindings/input/touchscreen/cypress%2Ctt21000.yaml#L86

is a remnant from the android cyttsp5 driver, which in turn is a remnant
of the android cyttsp4 driver.

The android driver has this if-statement where it basically supports
both level and falling edge interrupts:

https://github.com/aosp-rockchip/android_kernel_rockchip_rk356x/blob/8da04464e2bf51e45673d264c7a16e7309b8c793/drivers/input/touchscreen/cyttsp5/cyttsp5_core.c#L5962

However, when I grep for this level_irq_udelay variable, it is not used
anywhere in the cyttsp5 android driver. However, the cyttsp4 driver
actually uses it to add a delay to the interrupt handler, which makes
sense when you trigger on level low:

https://github.com/aosp-rockchip/android_kernel_rockchip_rk356x/blob/8da04464e2bf51e45673d264c7a16e7309b8c793/drivers/input/touchscreen/cyttsp4_core.c#L1238

Summarizing, I think the dt scheme and the pinenote dts should be fixed.
In case that you agree with my assessment, could you by chance advise me
how to proceed?
---
 arch/arm64/boot/dts/rockchip/rk3566-pinenote.dtsi | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm64/boot/dts/rockchip/rk3566-pinenote.dtsi b/arch/arm64/boot/dts/rockchip/rk3566-pinenote.dtsi
index cdb9fa6..50a20d5 100644
--- a/arch/arm64/boot/dts/rockchip/rk3566-pinenote.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3566-pinenote.dtsi
@@ -706,7 +706,7 @@ touchscreen@24 {
 		hid-descr-addr = <0x1>;
 		reg = <0x24>;
 		interrupt-parent = <&gpio0>;
-		interrupts = <RK_PA6 IRQ_TYPE_LEVEL_LOW>;
+		interrupts = <RK_PA6 IRQ_TYPE_EDGE_FALLING>;
 		pinctrl-0 = <&ts_int_l>, <&ts_rst_l>;
 		pinctrl-names = "default";
 		reset-gpios = <&gpio0 RK_PA5 GPIO_ACTIVE_LOW>;
